web/modules/weather_data/src/Service/Test/GetPlaceFromGrid.php.test

```diff
@@ -2,47 +2,78 @@
 
 namespace Drupal\weather_data\Service\Test;
 
-use GuzzleHttp\Psr7\Response;
+use Drupal\Core\Database\StatementInterface;
 
 /**
  * Test getting place information from the grid.
  */
 final class GetPlaceFromGridTest extends Base
 {
     /**
-     * Utility to quickly setup the happy path routes for this method.
+     * Tests the happy path.
      */
-    private function setupHappyPathRoutes()
+    public function testHappyPath(): void
     {
-        $this->httpClientMock->append(
-            new Response(
-                200,
-                ["Content-type" => "application/geo+json"],
-                '{"geometry":{"coordinates":[[[4,3],[5,9],[3,9]]]}}',
-            ),
+        $this->selfMock
+            ->method("getGeometryFromGrid")
+            ->will(
+                $this->returnValueMap([
+                    [
+                        "WFO",
+                        1,
+                        2,
+                        [
+                            (object) ["lat" => 7, "lon" => 8],
+                            (object) ["lat" => 9, "lon" => 10],
+                        ],
+                    ],
+                ]),
+            );
```

Comment:
So instead of mocking out an API HTTP response, we can instead directly mock the `getGeometryFromGrid` method.